keps/sig-storage/20190130-csi-raw-block-support.md

```diff
@@ -84,6 +87,14 @@ plugin removed any device files it created.
 
 This support is controlled by a feature gate called "CSIBlockVolume".
 
+## Upgrade/Downgrade Strategy
+
+CSIBlockVolume feature gate is used only in CSI volume plugin and only in
+the parts that are used by kubelet (`BlockVolumeMapper` interface). A node
+must be therefore drained before switching the feature off to remove all
```

Comment:
If the feature is off and there is a pod that uses the feature, the pod won't stop running and it still can use the device. When the pod is deleted, kubelet won't clean its volumes - all attempts will result in error that the volume plugin does not support block. Since it's just a symlink here and there, it won't harm the data, but it probably needs to be cleaned up manually.

Reply:
> If the feature is off and there is a pod that uses the feature, the pod won't stop running

At least in theory, I'll test it.

In any way, kubelet does not touch block devices, it only shuffles symlinks / losetup devices around.